src/use_case/MyAccountUseCase.java

```diff
@@ -35,7 +35,9 @@ public String getAllergies() {
     }
 
     public void updateAllergies(String allergies) {
-        userAccount.setAllergies(allergies);
+        if (!allergies.isEmpty()) {
+            userAccount.setAllergies(allergies);
```

Comment:
This line of code might not update when needing to clear the "allergies" variable.